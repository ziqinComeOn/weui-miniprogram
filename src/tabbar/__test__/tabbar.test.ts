import path from 'path'
import simulate from 'miniprogram-simulate'

describe('tabbar', () => {
    const tabbar = simulate.load(path.resolve(__dirname, '../tabbar'), 'mp-tabbar')

    test('basic', async () => {
        const tabChange = jest.fn()
        const tablist = [
            {
                text: '微信',
                iconPath: '/icon',
                selectedIconPath: '/icon.selected',
                badge: '8'
            },
            {
                text: '通讯录',
                iconPath: '/icon',
                selectedIconPath: '/icon.selected'
            },
            {
                text: '发现',
                iconPath: '/icon',
                selectedIconPath: '/icon.selected',
                dot: true
            },
            {
                text: '我',
                iconPath: '/icon',
                selectedIconPath: '/icon.selected'
            }
        ]
        const comp = simulate.render(
            simulate.load({
                usingComponents: {
                    'mp-tabbar': tabbar
                },
                template: `
                    <mp-tabbar id="tabbar" style="position:fixed;bottom:0;width:100%;left:0;right:0;" list="{{list}}" bindchange="tabChange"></mp-tabbar>
                `,
                data: {
                    list: tablist
                },
                methods: {
                    tabChange
                }
            })
        )
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        const tabbarComp = comp.querySelector('#tabbar')
        const tabs = tabbarComp.querySelectorAll('.weui-tabbar__item')
        tabs[2].dispatchEvent('tap')
        await simulate.sleep(0)
        expect(tabbarComp.data.current).toBe(2)
        expect(tabChange).toBeCalledTimes(1)
        expect(tabChange.mock.calls[0][0].detail).toStrictEqual({ index: 2, item: tablist[2] })
    })
})
