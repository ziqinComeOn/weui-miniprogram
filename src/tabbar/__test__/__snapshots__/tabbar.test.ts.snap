// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabbar basic 1`] = `
<main>
  <mp-tabbar
    id="tabbar"
    style="position:fixed;bottom:0;width:100%;left:0;right:0;"
    bind:change="tabChange"
  >
    <wx-view
      class="weui-tabbar "
    >
      <wx-view
        class="weui-tabbar__item weui-bar__item_on"
        data-index="{{0}}"
        bind:tap="tabChange"
      >
        <wx-view
          style="position: relative;display:inline-block;"
        >
          <wx-image
            class="weui-tabbar__icon"
            src="/icon.selected"
          />
          <mp-badge
            style="position: absolute;top:-2px;left:calc(100% - 3px)"
          >
            <wx-view
              class="weui-badge  "
            >
              8
            </wx-view>
          </mp-badge>
        </wx-view>
        <wx-view
          class="weui-tabbar__label"
        >
          微信
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-tabbar__item "
        data-index="{{1}}"
        bind:tap="tabChange"
      >
        <wx-view
          style="position: relative;display:inline-block;"
        >
          <wx-image
            class="weui-tabbar__icon"
            src="/icon"
          />
        </wx-view>
        <wx-view
          class="weui-tabbar__label"
        >
          通讯录
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-tabbar__item "
        data-index="{{2}}"
        bind:tap="tabChange"
      >
        <wx-view
          style="position: relative;display:inline-block;"
        >
          <wx-image
            class="weui-tabbar__icon"
            src="/icon"
          />
          <mp-badge
            style="position: absolute;top:-2px;left:calc(100% - 3px)"
          >
            <wx-view
              class="weui-badge  weui-badge_dot"
            >
              
            </wx-view>
          </mp-badge>
        </wx-view>
        <wx-view
          class="weui-tabbar__label"
        >
          发现
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-tabbar__item "
        data-index="{{3}}"
        bind:tap="tabChange"
      >
        <wx-view
          style="position: relative;display:inline-block;"
        >
          <wx-image
            class="weui-tabbar__icon"
            src="/icon"
          />
        </wx-view>
        <wx-view
          class="weui-tabbar__label"
        >
          我
        </wx-view>
      </wx-view>
    </wx-view>
  </mp-tabbar>
</main>
`;
